plugins/module_utils/client.py

```diff
@@ -115,39 +118,69 @@ def _get_resource(self, resource: Resource) -> Resource:
             json.loads(result["ResourceDescription"]["Properties"])
         )
 
+    @staticmethod
+    def make_result(changed, result, msg):
+        result = {"changed": changed, **result.resource}
+        if msg:
+            result["msg"] = msg
+        return result
+
     def _create(self, resource: Resource) -> Resource:
-        result = self.client.create_resource(
-            TypeName=resource.type_name, DesiredState=json.dumps(resource.properties)
-        )
-        try:
-            self._wait(result["ProgressEvent"]["RequestToken"])
-        except botocore.exceptions.WaiterError as e:
-            raise Exception(e.last_response["ProgressEvent"]["StatusMessage"])
-        return self._get_resource(resource)
+        changed = True
+        msg = ""
+        if self.module.check_mode:
+            msg = f"Would have created resource type {resource.type_name} if not in check_mode."
```

Comment:
Rather than having a message that's specific to check mode, why not just always include the operation that was, or would have, been done? So `msg` is always present and one of `Created`, `Updated`, `Deleted` or `Skipped` (no change).